tests/lib/rules/no-constant-condition.js

```diff
@@ -103,6 +112,16 @@ ruleTester.run("no-constant-condition", rule, {
         { code: "if(abc==='str' || true){}", errors: [{ message: "Unexpected constant condition.", type: "IfStatement" }] },
         { code: "if(abc==='str' || true || def ==='str'){}", errors: [{ message: "Unexpected constant condition.", type: "IfStatement" }] },
         { code: "if(false || true){}", errors: [{ message: "Unexpected constant condition.", type: "IfStatement" }] },
-        { code: "if(typeof abc==='str' || true){}", errors: [{ message: "Unexpected constant condition.", type: "IfStatement" }] }
+        { code: "if(typeof abc==='str' || true){}", errors: [{ message: "Unexpected constant condition.", type: "IfStatement" }] },
+
+        { code: "function* foo(){while(true){} yield 'foo';}", parserOptions: { ecmaVersion: 6 }, errors: [{ message: "Unexpected constant condition.", type: "WhileStatement" }] },
+        { code: "function* foo(){while(true){if (true) {yield 'foo';}}}", parserOptions: { ecmaVersion: 6 }, errors: [{ message: "Unexpected constant condition.", type: "IfStatement" }] },
+        { code: "function* foo(){while(true){yield 'foo';} while(true) {}}", parserOptions: { ecmaVersion: 6 }, errors: [{ message: "Unexpected constant condition.", type: "WhileStatement" }] },
+        { code: "var a = function* foo(){while(true){} yield 'foo';}", parserOptions: { ecmaVersion: 6 }, errors: [{ message: "Unexpected constant condition.", type: "WhileStatement" }] },
+        { code: "while (true) { function* foo() {yield;}}", parserOptions: { ecmaVersion: 6 }, errors: [{ message: "Unexpected constant condition.", type: "WhileStatement" }] },
+        { code: "function* foo(){if (true) {yield 'foo';}}", parserOptions: { ecmaVersion: 6 }, errors: [{ message: "Unexpected constant condition.", type: "IfStatement" }] },
+        { code: "function* foo() {for (let foo = yield; ;) {}}", parserOptions: { ecmaVersion: 6 }, errors: [{ message: "Unexpected constant condition.", type: "ForStatement" }] },
+        { code: "function foo() {while (true) {function* bar() {while (true) {yield;}}}}", parserOptions: { ecmaVersion: 6 }, errors: [{ message: "Unexpected constant condition.", type: "WhileStatement" }] }
```

Comment:
Nitpick: Can you space these tests out a bit more rather than having each test be a single line?

```js
{
    code: "function* foo(){while(true){} yield 'foo';}",
    errors: [{ message: "Unexpected constant condition.", type: "WhileStatement" }]
}
```

Personally, I think that makes the tests easier to read -- I find it a bit difficult to read the them when they're compacted like that.